File: python3/LeetCode/increasingBST.py
from TreeNode import TreeNode


class Solution(object):
    def increasingBST(self, root):
        dummy = TreeNode(-1)
        self.res = dummy
        self.BST(root);
        return dummy.right

    def BST(self, root):
        if root != None:
            self.BST(root.left)
            self.res.right = root
            root.left = None
            self.res = root
            self.BST(root.right)

if __name__ == '__main__':
    node1 = TreeNode(5)
    node2 = TreeNode(1)
    node3 = TreeNode(7)
    node1.left = node2
    node1.right = node3
    print(Solution().increasingBST(node1))

File: go/LeetCode/increasingBST.go
package main

import "fmt"

//type TreeNode struct {
//	Val int
//	Left *TreeNode
//	Right *TreeNode
//}

func main() {
	node1 := &TreeNode{}
	node1.Val = 5
	node2 := &TreeNode{}
	node2.Val = 1
	node3 := &TreeNode{}
	node3.Val = 7

	node1.Left  = node2
	node1.Right  = node3

	fmt.Print(increasingBST(node1))
}

func increasingBST(root *TreeNode) *TreeNode {
	dummy :=&TreeNode{}
	res := dummy

	var BST func(*TreeNode)
	BST = func(node *TreeNode) {
		if node != nil {
			BST(node.Left)
			res.Right = node
			node.Left = nil
			res = node
			BST(node.Right)
		}
	}
	BST(root)
	return dummy.Right
}
